File: benchmarks/xlang/src/wordfreq.R
n <- 3000000
t0 <- proc.time()[["elapsed"]]
m <- new.env(hash = TRUE)
seed <- 42
for (i in 1:n) {
  seed <- (seed * 48271) %% 2147483647
  word <- paste0("w", seed %% 50000)
  cur <- m[[word]]
  if (is.null(cur)) m[[word]] <- 1L else m[[word]] <- cur + 1L
}
maxf <- 0L
distinct <- 0L
for (k in ls(m)) {
  distinct <- distinct + 1L
  v <- m[[k]]
  if (v > maxf) maxf <- v
}
cat("CHECK", format(distinct * 1000000 + maxf, scientific = FALSE), "\n")
cat("MS", as.integer((proc.time()[["elapsed"]] - t0) * 1000), "\n")

File: benchmarks/xlang/src/Wordfreq.java
import java.util.HashMap;
public class Wordfreq {
    public static void main(String[] a) {
        final int n = 3000000;
        long t0 = System.nanoTime();
        HashMap<String, Long> m = new HashMap<>();
        long seed = 42;
        for (int i = 0; i < n; i++) {
            seed = seed * 48271 % 2147483647L;
            m.merge("w" + (seed % 50000), 1L, Long::sum);
        }
        long maxf = 0;
        for (long v : m.values()) if (v > maxf) maxf = v;
        System.out.println("CHECK " + (m.size() * 1000000L + maxf));
        System.out.println("MS " + (System.nanoTime() - t0) / 1000000);
    }
}
